specification/communication/data-plane/PhoneNumbers/preview/2021-10-25-preview/phonenumbers.json

```diff
@@ -0,0 +1,1639 @@
+            "schema": {
+              "$ref": "#/definitions/PhoneNumberSearchResult"
+            },
+            "headers": {
+              "Location": {
+                "description": "URL to retrieve the final result after operation completes.",
+                "type": "string"
+              },
+              "Operation-Location": {
+                "description": "URL to query for status of the operation.",
+                "type": "string"
+              },
+              "operation-id": {
+                "description": "The operation id.",
+                "type": "string"
+              },
+              "search-id": {
+                "description": "The search operation id.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Failure",
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "Search for phone numbers": {
+            "$ref": "./examples/SearchPhoneNumbers.json"
+          }
+        }
+      }
+    },
+    "/availablePhoneNumbers/searchResults/{searchId}": {
+      "get": {
+        "tags": [
+          "Find and purchase available phone numbers"
+        ],
+        "summary": "Gets a phone number search result by search id.",
+        "operationId": "PhoneNumbers_GetSearchResult",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "searchId",
+            "description": "The search Id.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/PhoneNumberSearchResult"
+            }
+          },
+          "default": {
+            "description": "Failure",
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get PhoneNumber Search Result": {
+            "$ref": "./examples/GetPhoneNumberSearchResult.json"
+          }
+        }
+      }
+    },
+    "/availablePhoneNumbers/:purchase": {
```

Comment:
Any reason not to take the `searchId` in the path since it looks like you're only allowed to buy the entire lot or not?

Reply:
This section of the swagger is also part existing endpoints that have already been GAd and are not changing as part of this new version.

Regarding this particular question, I'm not sure what was the rationale for preferring to put the searchId within the request body. I suppose that the endpoint would have needed to be modeled a little different like POST /availablePhoneNumbers/searchResults/{searchId}/:purchase and maybe the person who worked on that design did not find it desirable to have empty body request or model the purchase operation too tight to the concept of a search but I'm just guessing.

Reply:
We hadn't put the searchId in the path to allow for API evolution with other purchase options in the future. This way a new API version could introduce a new property in the request body to directly purchase a number instead of a search id, for example.